cmd/dockerd/daemon.go

```diff
@@ -861,37 +860,31 @@ func systemContainerdRunning(honorXDG bool) (string, bool, error) {
 	return addr, err == nil, nil
 }
 
-// configureDaemonLogs sets the logrus logging level and formatting. It expects
+// configureDaemonLogs sets the logging level and formatting. It expects
 // the passed configuration to already be validated, and ignores invalid options.
 func configureDaemonLogs(conf *config.Config) {
-	if conf.LogLevel != "" {
-		lvl, err := logrus.ParseLevel(conf.LogLevel)
-		if err == nil {
-			logrus.SetLevel(lvl)
-		}
-	} else {
-		logrus.SetLevel(logrus.InfoLevel)
-	}
-	logFormat := conf.LogFormat
-	if logFormat == "" {
-		logFormat = log.TextFormat
-	}
-	var formatter logrus.Formatter
-	switch logFormat {
+	switch conf.LogFormat {
 	case log.JSONFormat:
-		formatter = &logrus.JSONFormatter{
-			TimestampFormat: jsonmessage.RFC3339NanoFixed,
-		}
-	case log.TextFormat:
-		formatter = &logrus.TextFormatter{
-			TimestampFormat: jsonmessage.RFC3339NanoFixed,
+		logrus.SetFormatter(&logrus.JSONFormatter{
+			TimestampFormat: log.RFC3339NanoFixed,
+		})
+	case log.TextFormat, "":
+		logrus.SetFormatter(&logrus.TextFormatter{
+			TimestampFormat: log.RFC3339NanoFixed,
 			DisableColors:   conf.RawLogs,
 			FullTimestamp:   true,
-		}
+		})
 	default:
-		panic("unsupported log format " + logFormat)
+		panic("unsupported log format " + conf.LogFormat)
+	}
+
+	logLevel := conf.LogLevel
+	if logLevel == "" {
+		logLevel = "info"
+	}
+	if err := log.SetLevel(logLevel); err != nil {
+		log.G(context.TODO()).WithError(err).Warn("configure log level")
 	}
```

Comment:
If you're wondering what changed here; I changed the order of processing `logLevel` and `logFormat`; previously it would first parse the level, then parse the _format_ (which could "panic" if invalid).

However, that also meant that the "invalid log level" error would always be logged in the default format (not the one specified in `--log-format`).

So I swapped the order.

All of the above said; we should consider adding an error-return here, and make it a hard failure, instead of "log an error and continue".